apps/src/lib/client/rpc.rs

```diff
@@ -70,18 +126,988 @@ pub async fn query_balance(args: args::QueryBalance) {
                 let balances =
                     query_storage_prefix::<token::Amount>(client.clone(), key)
                         .await;
-                writeln!(w, "Token {}:", currency_code).unwrap();
-                for (key, balance) in balances {
-                    let owner = token::is_any_token_balance_key(&key).unwrap();
-                    writeln!(w, "  {}, owned by {}", balance, owner).unwrap();
+                match balances {
+                    Some(balances) => {
+                        writeln!(w, "Token {}:", currency_code).unwrap();
+                        for (key, balance) in balances {
+                            let owner =
+                                token::is_any_token_balance_key(&key).unwrap();
+                            writeln!(w, "  {}, owned by {}", balance, owner)
+                                .unwrap();
+                        }
+                    }
+                    None => {
+                        println!("No balances for token {}", token.encode())
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Query PoS bond(s)
+pub async fn query_bonds(args: args::QueryBonds) {
+    let epoch = query_epoch(args.query.clone()).await;
+    if let Some(epoch) = epoch {
+        let client = HttpClient::new(args.query.ledger_address).unwrap();
+        match (args.owner.as_ref(), args.validator.as_ref()) {
+            (Some(owner), Some(validator)) => {
+                // Find owner's delegations to the given validator
+                let bond_id = pos::BondId {
```

Comment:
This isn't really important, but it hurts me to see a clone of data values that are only used behind immutable references later.

Reply:
good point, fixed in ef33286